diagram.md

```diff
@@ -2,240 +2,284 @@
+|        |            | Audio        | SamplingFrequency |
+|        |            |              |-------------------|
+|        |            |              | Channels          |
+|        |            |              |-------------------|
+|        |            |              | BitDpeth          |
+|--------------------------------------------------------|
+
+```
+Figure: Representation of the `Tracks Element` and a selection of its `Descendant Elements`.
+
+The `Chapters Element` section lists all of the Chapters. Chapters are a way to set predefined points to jump to in video or audio.
+
+```
++----------------------------------------------+
+| Chatpers | EditionEntry | EditionUID         |
+|          |              |--------------------|
+|          |              | EditionFlagHidden  |
+|          |              |--------------------|
+|          |              | EditionFlagDefault |
+|          |              |--------------------|
+|          |              | EditionFlagOrdered |
+|          |              |--------------------|-------------------+
+|          |              | ChapterAtom        | ChapterUID        |
+|          |              |                    |-------------------|
+|          |              |                    | ChapterStringUID  |
+|          |              |                    |-------------------|
+|          |              |                    | ChapterTimeStart  |
+|          |              |                    |-------------------|
+|          |              |                    | ChapterTimeEnd    |
+|          |              |                    |-------------------|
+|          |              |                    | ChapterFlagHidden |
+|          |              |                    |-------------------|--------------+
+|          |              |                    | ChapterDisplay    | ChapString   |
+|          |              |                    |                   |--------------| 
+|          |              |                    |                   | ChapLanguage | 
++---------------------------------------------------------------------------------+
+```
+Figure: Representation of the `Chapters Element` and a selection of its `Descendant Elements`.
+
+The `Cluster Elements` contain all of the video frames and audio for each track. In a given Matroska file, there are usually many `Cluster Elements`. The Clusters help to break up the `SimpleBlock` or `BlockGroup Elements` and help with seeking and error protection. There is no set limit to how much data a Cluster can contain, or how much time they can span, but a limit at 5 seconds or 5 megabytes is RECOMMENDED. Every Cluster contains a timecode, usually the timecode that the first Block in the Cluster SHOULD be played back, but it doesn't have to be. Then there are one or more (usually many more) `BlockGroups` or `SimpleBlocks` in each Cluster. A BlockGroup can contain a Block of data, and any information relating directly to that Block.
```

Comment:
It's not quite correct that "there is no set limit…". One limit is implied by the fact that the `BlockGroup` and `SimpleBlock` elements use a 16bit signed integer to represent the relative timestamp (relative to the cluster's `ClusterTimestamp` element), scaled by the track's `TimecodeScale` element. Therefore the amount of time one can place in a cluster is definitely limited. For example, with the default resolution of 1ms you can only store ~32s of data (~65s if you use negative relative values, too).